### PR TITLE
Update mongo-db/mongoose dependents

### PR DESCRIPTION
**Pull request by sandersn**
Both mongodb and mongoose ship their own types now, and only the latest versions compile on TS 4.7 without errors.

This required a few changes to packages:

- A few tests were out of date. I put in declarations of mongoose/mongodb types in place of calls to out-of-date functions.
- mongoose-simple-random referred to an out-of-date interface and a no-longer-required type parameter constraint. I deleted both.
- mongorito needed to update a couple of the mongodb option types for createIndex and dropIndex

mongoose-simple-random is the biggest change overall.

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## mongoose-autopopulate ([<kbd>unpkg</kbd>](https://unpkg.com/browse/mongoose-autopopulate@latest/))
was missing the following properties:
1. The declaration doesn't match the JavaScript module 'mongoose-autopopulate'. Reason:
The JavaScript module can be called or constructed, but the declaration module cannot.

The most common way to resolve this error is to use 'export =' syntax.
To learn more about 'export =' syntax, see https://www.typescriptlang.org/docs/handbook/modules.html#export--and-import--require.
## mongoose-deep-populate ([<kbd>unpkg</kbd>](https://unpkg.com/browse/mongoose-deep-populate@latest/))
was missing the following properties:
1. The declaration doesn't match the JavaScript module 'mongoose-deep-populate'. Reason:
The JavaScript module can be called or constructed, but the declaration module cannot.

The most common way to resolve this error is to use 'export =' syntax.
To learn more about 'export =' syntax, see https://www.typescriptlang.org/docs/handbook/modules.html#export--and-import--require.
## mongoose-paginate ([<kbd>unpkg</kbd>](https://unpkg.com/browse/mongoose-paginate@latest/))
was missing the following properties:
1. paginate
## mongorito ([<kbd>unpkg</kbd>](https://unpkg.com/browse/mongorito@latest/))
was missing the following properties:
1. The declaration doesn't match the JavaScript module 'mongorito'. Reason:
The JavaScript module can be called or constructed, but the declaration module cannot.

The most common way to resolve this error is to use 'export =' syntax.
To learn more about 'export =' syntax, see https://www.typescriptlang.org/docs/handbook/modules.html#export--and-import--require.
## passport-local-mongoose ([<kbd>unpkg</kbd>](https://unpkg.com/browse/passport-local-mongoose@latest/))
was missing the following properties:
1. errors
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against f3dc38fa7c9323abdc7912cd8ccc91ca01643bc8
</p>

File: types/mongoose-deep-populate/mongoose-deep-populate-tests.ts [hunk 1]
```diff
@@ -2,15 +2,16 @@ import mongooseDeepPopulate from 'mongoose-deep-populate';
 import * as mongoose from 'mongoose';
 import { Schema } from 'mongoose';
 
-var connection = mongoose.connect("mongodb://localhost/myDatabase");
+async function wrapper() {
+    var connection = await mongoose.connect("mongodb://localhost/myDatabase");
+    var deepPopulate = mongooseDeepPopulate(connection);
 
-var deepPopulate = mongooseDeepPopulate(connection);
+    var bookSchema = new Schema({
```
Comment (isaacdecoded) [inline types/mongoose-deep-populate/mongoose-deep-populate-tests.ts:9]: Excuse me, it's probably a superfluous catch but, can _vars_ be replaced by _consts_?

**Review by isaacdecoded — APPROVED** [review]
Changes LGTM, thanks.

💛

**Review by weswigham — APPROVED** [review]

**Review by bondz — APPROVED** [review]

**Review by kerolloz — APPROVED** [review]
Thank you for making this PR 💯

**Comment by peterblazejewicz** [comment]
FYI: merging this after asking on Discord to push solving DT problems forward skipping self-merge by OP/PR owner:
https://discord.com/channels/508357248330760243/757992230077333655/967785395200864267

**Comment by peterblazejewicz** [comment]
I've tested this on my own #59796 and it fixes the CI issue for @types/node changes:
https://github.com/DefinitelyTyped/DefinitelyTyped/runs/6147223688?check_suite_focus=true

**Pull request merged**